code/game/gamemodes/changeling/implements/items.dm

```diff
@@ -122,3 +122,163 @@
+	user.death() // double tap - geeves
+
+/obj/item/projectile/tentacle
+	name = "tentacle"
+	icon = 'icons/obj/contained_items/weapons/ling_tentacle.dmi'
+	icon_state = "tentacle_end"
+	pass_flags = PASSTABLE
+	cant_miss = TRUE
+	damage = 0
+	damage_type = BRUTE
+	range = 8
+	hitsound = 'sound/weapons/thudswoosh.ogg'
+	var/chain
+	var/obj/item/gun/energy/tentacle/source //the item that shot it
+
+/obj/item/projectile/tentacle/New(obj/item/gun/energy/tentacle/tentacle_gun)
+	source = tentacle_gun
+	..()
+
+/obj/item/projectile/tentacle/fire(setAngle)
+	if(firer)
+		chain = firer.Beam(src, icon_state = "tentacle", time = -1, maxdistance = INFINITY, beam_sleep_time = 1)
+	..()
+
+/obj/item/projectile/tentacle/proc/reset_throw(mob/living/carbon/human/H)
+	if(H.in_throw_mode)
+		H.throw_mode_off() //Don't annoy the changeling if he doesn't catch the item
+
+/mob/proc/tentacle_grab(mob/living/carbon/human/H)
+	if(ishuman(H) && Adjacent(H))
+		var/obj/item/grab/G = H.grabbedby(src, TRUE)
+		if(istype(G))
+			G.set_state(GRAB_AGGRESSIVE) //Instant aggressive grab
+
+/mob/proc/tentacle_stab(mob/living/carbon/C)
+	if(Adjacent(C))
+		var/obj/item/I = r_hand
+		if(!is_sharp(I))
+			I = l_hand
+		if(!is_sharp(I))
+			return
+
+		C.visible_message(SPAN_DANGER("[src] impales [C] with [I]!"), SPAN_DANGER("[src] impales you with [I]!"))
+		C.apply_damage(I.force, BRUTE, "chest")
+		do_attack_animation(C)
+		add_blood(C)
+		playsound(get_turf(src), I.hitsound, 75, 1)
+
+/obj/item/projectile/tentacle/on_hit(atom/target, blocked = 0)
+	qdel(source) //one tentacle only unless you miss
+	if(blocked >= 100)
+		return FALSE
+	var/mob/living/carbon/human/H = firer
+	if(istype(target, /obj/item))
+		var/obj/item/I = target
+		if(!I.anchored)
+			to_chat(firer, SPAN_NOTICE("You pull [I] towards yourself."))
+			H.throw_mode_on()
+			I.throw_at(H, 10, 2)
+			. = TRUE
+
+	else if(isliving(target))
+		var/mob/living/L = target
+		if(!L.anchored && !L.throwing)//avoid double hits
+			if(iscarbon(L))
+				var/mob/living/carbon/C = L
+				switch(firer.a_intent)
+					if(I_HELP)
+						C.visible_message(SPAN_DANGER("[L] is pulled by [H]'s tentacle!"), SPAN_DANGER("A tentacle grabs you and pulls you towards [H]!"))
+						C.throw_at(get_step_towards(H,C), 8, 2)
+						return TRUE
+
+					if(I_DISARM)
+						var/obj/item/I = C.get_active_hand()
+						if(!I || istype(I, /obj/item/grab))
+							I = C.get_inactive_hand()
+						if(I && !istype(I, /obj/item/grab))
+							if(C.unEquip(I))
+								C.visible_message(SPAN_DANGER("[I] is yanked out of [C]'s hand by [src]!"),SPAN_DANGER("A tentacle pulls [I] away from you!"))
+								I.throw_at(get_step_towards(H, C), 8, 2, callback = CALLBACK(H, /mob/proc/put_in_any_hand_if_possible, I))
+								return TRUE
+							else
+								to_chat(firer, SPAN_DANGER("You can't seem to pry [I] out of [C]'s hands!"))
+								return FALSE
+						else
+							to_chat(firer, SPAN_DANGER("[C] has nothing in hand to disarm!"))
+							return FALSE
```

Comment:
I have concerns about that capability, as we removed a lot of the "slap things out of hands" mechanics in the past due to balance issues.